server/api/graphql_root_run.go

```diff
@@ -179,12 +179,37 @@ func (r *RunRootResolver) UpdateRun(ctx context.Context, args struct {
 	setmap := map[string]interface{}{}
 	addToSetmap(setmap, "Name", args.Updates.Name)
 	addToSetmap(setmap, "Description", args.Updates.Summary)
-	addToSetmap(setmap, "ChannelID", args.Updates.ChannelID)
 	addToSetmap(setmap, "CreateChannelMemberOnNewParticipant", args.Updates.CreateChannelMemberOnNewParticipant)
 	addToSetmap(setmap, "RemoveChannelMemberOnRemovedParticipant", args.Updates.RemoveChannelMemberOnRemovedParticipant)
 	addToSetmap(setmap, "StatusUpdateBroadcastChannelsEnabled", args.Updates.StatusUpdateBroadcastChannelsEnabled)
 	addToSetmap(setmap, "StatusUpdateBroadcastWebhooksEnabled", args.Updates.StatusUpdateBroadcastWebhooksEnabled)
 
+	if args.Updates.ChannelID != nil {
+		channel, err := c.pluginAPI.Channel.Get(*args.Updates.ChannelID)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to get channel")
+		}
+
+		if channel.TeamId != playbookRun.TeamID {
+			return "", errors.Wrap(app.ErrMalformedPlaybookRun, "channel not in given team")
+		}
+
+		permission := model.PermissionManagePublicChannelProperties
+		permissionMessage := "You are not able to manage public channel properties"
+		if channel.Type == model.ChannelTypePrivate {
+			permission = model.PermissionManagePrivateChannelProperties
+			permissionMessage = "You are not able to manage private channel properties"
+		} else if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
```

Comment:
nit: suggest using `channel.IsGroupOrDirect()`, I just learned about this :p
```suggestion
		} else if channel.IsGroupOrDirect() {
```